paddlenlp/transformers/model_utils.py

```diff
@@ -44,6 +46,38 @@
 ]
 
 
+def get_path_from_url(url: str,
+                      root_dir: str,
+                      md5sum: Optional[str] = None,
+                      check_exist: bool = True) -> str:
+    """construct `get_path_from_url` for `model_utils` to avoid multi-process downloading
+
+    Args:
+        url (str): the url of resource file
+        root_dir (str): the local download path
+        md5sum (str, optional): md5sum string for file. Defaults to None.
+        check_exist (bool, optional): whether check the file is exist. Defaults to True.
+
+    Returns:
+        str: the path of downloaded file
+    """
+    os.makedirs(root_dir, exist_ok=True)
+    lock_file = os.path.join(root_dir, '.download_lock_file')
+
+    # create lock file, which is empty, under the `LOCK_FILE_HOME` directory.
+    lock_file = os.path.join(LOCK_FILE_HOME, str(hash(url + root_dir)))
+    with FileLock(lock_file) as file_lock:
+        file_lock.acquire()
```

Comment:
I don't think this is right? 
since we already created the lock at line 69, `file_lock.acquire()` is just going to hang

Reply:
yes, this code block has been updated in new commit.  it's actually a bug  but fixed.